chia/consensus/blockchain.py

```diff
@@ -761,10 +757,8 @@ async def get_header_blocks_in_range(
         hashes = []
         for height in range(start, stop + 1):
             if self.contains_height(uint32(height)):
-                # TODO: address hint error and remove ignore
-                #       error: Incompatible types in assignment (expression has type "Optional[bytes32]", variable has
-                #       type "bytes32")  [assignment]
-                header_hash: bytes32 = self.height_to_hash(uint32(height))  # type: ignore[assignment]
+                header_hash: Optional[bytes32] = self.height_to_hash(uint32(height))
+                assert header_hash is not None
```

Comment:
Would this have resulted in an exception before? Or just not gotten a record from the db?  I looked down to the query but haven't traced back up the call stack to see the effect of an incomplete result.

Reply:
To be explored separately.

Reply:
https://github.com/Chia-Network/chia-blockchain/pull/11007